common/src/main/java/com/wynntils/screens/partymanagement/PartyManagementScreen.java

```diff
@@ -0,0 +1,426 @@
+                inviteButton = new Button.Builder(
+                                Component.translatable("screens.wynntils.partyManagementGui.invite"),
+                                (button) -> inviteFromField())
+                        .pos(this.width / 2 - (X_START - TOTAL_WIDTH + 40), this.height / 2 - 200)
+                        .size(40, 20)
+                        .build());
+        // endregion
+
+        // region Management button row
+        this.addRenderableWidget(new Button.Builder(
+                        Component.translatable("screens.wynntils.partyManagementGui.refreshButton")
+                                .withStyle(ChatFormatting.GREEN),
+                        (button) -> refreshParty())
+                .pos(this.width / 2 - X_START, this.height / 2 - 176)
+                .size(83, 20)
+                .build());
+        this.addRenderableWidget(
+                kickOfflineButton = new Button.Builder(
+                                Component.translatable("screens.wynntils.partyManagementGui.kickOfflineButton")
+                                        .withStyle(ChatFormatting.RED),
+                                (button) -> kickOffline())
+                        .pos(this.width / 2 - (X_START - 87), this.height / 2 - 176)
+                        .size(83, 20)
+                        .build());
+        this.addRenderableWidget(
+                createPartyButton = new Button.Builder(
+                                Component.translatable("screens.wynntils.partyManagementGui.createPartyButton"),
+                                (button) -> createParty())
+                        .pos(this.width / 2 - (X_START - 174), this.height / 2 - 176)
+                        .size(83, 20)
+                        .build());
+        this.addRenderableWidget(
+                leavePartyButton = new Button.Builder(
+                                Component.translatable("screens.wynntils.partyManagementGui.leavePartyButton")
+                                        .withStyle(ChatFormatting.RED),
+                                (button) -> leaveParty())
+                        .pos(this.width / 2 - (X_START - 261), this.height / 2 - 176)
+                        .size(83, 20)
+                        .build());
+        // endregion
+    }
+
+    @Override
+    public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        renderBackground(poseStack);
+        super.render(poseStack, mouseX, mouseY, partialTick);
+
+        boolean partying = Models.Party.isPartying();
+
+        createPartyButton.active = !partying;
+        leavePartyButton.active = partying;
+        kickOfflineButton.active = partying && !offlineMembers.isEmpty();
+        inviteButton.active = !inviteInput
+                .getTextBoxInput()
+                .isBlank(); // partying check not required as button automatically makes new party if not in one
+        refreshOffline();
```

Comment:
We are still doing this render time. I will need to clone this whole PR to check how to work around this.